### PR TITLE
feat: use the new domain scheme for exposed services

### PR DESCRIPTION
**Pull request by utkuozdemir**
Rework the workload service proxying feature to support the new domain format in addition to the existing one.

Old format:
```
p-g3a4ana-demo.omni.siderolabs.io
```

New format:
```
g3a4ana-demo.proxy-us.omni.siderolabs.io
```

The old format required a new DNS records to be added for each new workload service, causing issues with the resolution on clients. The new format addresses it by leveraging wildcard records.

Additionally, build the full exposed service URL on the backend and make it a field on `ExposedService` resource, so they can be accessed using `omnictl get exposedservice`.

Part of siderolabs/omni#17.

### CODE REVIEW

File: internal/backend/runtime/omni/controllers/omni/kubernetes_status.go [hunk 1]
```diff
@@ -240,6 +257,29 @@ func (ctrl *KubernetesStatusController) updateExposedServices(ctx context.Contex
...
+	restParts := strings.SplitN(rest, ".", 2)
+	if len(restParts) != 2 {
+		return "", fmt.Errorf("invalid advertised API URL: %s", ctrl.advertisedAPIURL)
+	}
+
+	instanceName := restParts[0]
+	rest = restParts[1]
+
+	// example: p-g3a4ana-demo.proxy-us.omni.siderolabs.io
+	url := protocol + "//" + workloadproxy.HostPrefix + "-" + alias + "-" + instanceName + "." + ctrl.workloadProxySubdomain + "." + rest
```
Comment (Unix4ever) [inline internal/backend/runtime/omni/controllers/omni/kubernetes_status.go:278]: Do we still use `p-` to route the requests?

Comment (utkuozdemir) [reply internal/backend/runtime/omni/controllers/omni/kubernetes_status.go:278]: Actually, good point, we don't need it anymore. The instance name is still good to ensure there are no collisions, but `p-` is not needed really. Should we just drop it?

Comment (Unix4ever) [reply internal/backend/runtime/omni/controllers/omni/kubernetes_status.go:278]: Could be cool to provide a way to also make alias static, to get even prettier URL :)

Comment (Unix4ever) [reply internal/backend/runtime/omni/controllers/omni/kubernetes_status.go:278]: But I guess we should definitely drop `p-` prefix

Comment (utkuozdemir) [reply internal/backend/runtime/omni/controllers/omni/kubernetes_status.go:278]: We had something like that as the initial idea back then, but there was a problem with potential collisions.

If it's ok, I'd leave that for another iteration to make sure we get it right if we do it again. There are some other topics we might want to address in a next iteration.

I'd say let's address the DNS propagation issue in this one. The changes we would add later would still use the new subdomain introduced here anyway.

But I'm gonna drop the `p-` - it's definitely not needed.

Comment (utkuozdemir) [reply internal/backend/runtime/omni/controllers/omni/kubernetes_status.go]: dropped `p-`, also updated the omni-controller.

File: internal/backend/workloadproxy/handler.go [hunk 2]
```diff
@@ -218,3 +245,16 @@ func getMainDomain(url *url.URL) string {
 
 	return url.Host
 }
+
+// getWorkloadProxyDomain returns the full domain used by the workload proxy as the parent domain.
+//
+// Example: proxy-us.omni.siderolabs.io.
+func getWorkloadProxyDomain(subdomain string, mainDomain string) string {
+	parts := strings.SplitN(mainDomain, ".", 2)
```
Comment (DmitriyMV) [inline internal/backend/workloadproxy/handler.go:253]:
```suggestion
	left, right, ok := strings.Cut(mainDomain, ".")
```

File: frontend/src/api/resources.ts [hunk 3]
```diff
@@ -21,13 +21,13 @@ export const authBearerHeaderPrefix = "Bearer ";
 export const SignatureVersionV1 = "siderov1";
 export const samlSessionHeader = "saml-session";
 export const DefaultKubernetesVersion = "1.30.1";
+export const workloadProxyHostPrefix = "p";
```
Comment (DmitriyMV) [inline frontend/src/api/resources.ts:24]: Do we need this const?

Comment (utkuozdemir) [reply frontend/src/api/resources.ts:24]: ah, I removed the `tsgen:` hint but forgot to re-generate afterwards.

**Review by Unix4ever — APPROVED** [review]

**Pull request merged**